cmd/vic-machine/create/create.go

```diff
@@ -176,6 +198,166 @@ func (c *Create) Flags() []cli.Flag {
+	}
+
+	vnet := mapped[0]
+	if vnet == "" {
+		if vnet == "" {
+			return "", nil, cli.NewExitError(fmt.Sprintf("Container network not specified"), 1)
+		}
+	}
+
+	gw, subnet, err := net.ParseCIDR(mapped[1])
+	if err != nil {
+		return "", nil, cli.NewExitError(fmt.Sprintf("Invalid gateway %s specified for container network %s", mapped[1], vnet), 1)
+	}
+
+	return vnet, &net.IPNet{IP: gw, Mask: subnet.Mask}, nil
+}
+
+func (c *Create) processContainerNetworkGateways() (map[string]net.IPNet, error) {
+	gws := make(map[string]net.IPNet)
+	for _, cg := range c.containerNetworksGateway {
+		vnet, gw, err := parseGatewaySpec(cg)
+		if err != nil {
+			return nil, err
+		}
+
+		if _, ok := gws[vnet]; ok {
+			return nil, cli.NewExitError(fmt.Sprintf("Duplicate gateway specified for container network %s", vnet), 1)
+		}
+
+		gws[vnet] = *gw
+	}
+
+	return gws, nil
+}
+
+func parseIPRangeSpec(cp string) (vnet string, ipr metadata.IPRange, err error) {
+	if cp == "" {
+		err = cli.NewExitError("container network ip range is empty", 1)
+		return
+	}
+
+	mapped := strings.Split(cp, ":")
+	if len(mapped) != 2 {
+		err = cli.NewExitError(fmt.Sprintf("Container network ip range %s not in VSPHERE-NETWORK:IP-RANGE format", cp), 1)
+		return
+	}
+
+	vnet = mapped[0]
+	if vnet == "" {
+		err = cli.NewExitError(fmt.Sprintf("Container network not specified"), 1)
+		return
+	}
+
+	r := metadata.ParseIPRange(mapped[1])
+	if r == nil {
+		err = cli.NewExitError(fmt.Sprintf("Invalid ip range for container network %s", vnet), 1)
+		return
+	}
+
+	ipr = *r
+	return
+}
+
+func (c *Create) processContainerNetworkIPRanges() (map[string][]metadata.IPRange, error) {
+	pools := make(map[string][]metadata.IPRange)
+	for _, cp := range c.containerNetworksIPRanges {
+		vnet, ipr, err := parseIPRangeSpec(cp)
+		if err != nil {
+			return nil, err
+		}
+
+		p := pools[vnet]
+		if p == nil {
+			p = []metadata.IPRange{ipr}
+		} else {
+			// ip range should not overlap with any other range
+			for _, i := range p {
+				if i.Overlaps(ipr) {
+					return nil, cli.NewExitError(fmt.Sprintf("Overlapping IP ranges: %s %s", i, ipr), 1)
+				}
+			}
+			p = append(p, ipr)
+		}
+
+		pools[vnet] = p
+	}
+
+	return pools, nil
+}
+
+func (c *Create) processContainerNetworks() error {
+	gws, err := c.processContainerNetworkGateways()
+	if err != nil {
+		return err
+	}
+
+	pools, err := c.processContainerNetworkIPRanges()
+	if err != nil {
+		return err
+	}
+
+	// parse container networks
+	for _, cn := range c.containerNetworks {
+		if cn == "" {
+			return cli.NewExitError("Container network mapping not specified", 1)
+		}
+
+		mapped := strings.Split(cn, ":")
+		if len(mapped) > 2 {
+			return cli.NewExitError(fmt.Sprintf("Container network mapping %s not in VSPHERE-NETWORK[:VIC-NETWORK] format", cn), 1)
+		}
+
+		vnet := mapped[0]
+		vicnet := vnet
+		if len(mapped) > 1 {
+			vicnet = mapped[1]
+		}
+
+		if vicnet == "bridge" {
```

Comment:
how do we specify IP range/gateway for bridge, and how do we avoid conflict with other addresses? Do we need to calculate a non-conflicting scope automatically?